From c121da9492b81a9a7572ffb30ccb7252624193f6 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 14 Apr 2022 10:51:53 +0300
Subject: [PATCH] [SW-82943] gaudi3: read and store syscfg2 offset

In gaudi3, ROM need information fuse information stored in 58
as well 59 offset. The 59 offset which is named as syscfg2 holds
information of D2D and PLL values. This patch add change to
read and store 59 fuse offset in real and fuse emulation
scenario.

Change-Id: I4663dcc85d171eabe9e1966759ba97aa762da59e
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 8ebe782..be4e6a5 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -28,6 +28,7 @@
 
 struct arcboot_global_s arcboot_rom_global = {
 	.config_fuse = 0,
+	.config2_fuse = 0,
 	.recovery_flag = 0,
 	.validation_pass = 0,
 	.efuse_read = 0,
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index de1d0ab..19d1a5c 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -401,6 +401,7 @@
 		val = EFUSE_SYSCFG_DEFAULT;
 
 	arcboot_rom_global.config_fuse = val;
+	fuse_read(EFUSE_SYSCFG2_OFFSET, &arcboot_rom_global.config2_fuse);
 
 	/* Override the fuse values as per requirement */
 	arcboot_rom_fuse_override();
diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index 9edb8a1..71a9f99 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -8,6 +8,7 @@
 
 /* Unused for gaudi3. Maintained for common code compilation */
 #define EFUSE_KEY_HASH_OFFSET				0x0
+#define EFUSE_SYSCFG2_OFFSET				0x3B
 
 #define EFUSE_SYSCFG_REVOKE_KEY_0			BIT(0)
 #define EFUSE_SYSCFG_PERMIT_KEY_4			~BIT(4)
diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index 99d8659..372b8b1 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -56,5 +56,6 @@
 		val = EFUSE_SYSCFG_DEFAULT;
 
 	arcboot_rom_global.config_fuse = val;
+	fuse_read(EFUSE_SYSCFG2_OFFSET, &arcboot_rom_global.config2_fuse);
 	arcboot_rom_status_update(PROGRESS_INIT_EFUSE_SPI);
 }
diff --git a/include/arcboot.h b/include/arcboot.h
index 7a013ed..ccdeb07 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -76,6 +76,7 @@
 
 struct arcboot_global_s {
 	uint32_t config_fuse;
+	uint32_t config2_fuse;
 	uint8_t recovery_flag;
 	uint8_t validation_pass;
 	uint8_t efuse_read;
